src/index.js

```diff
@@ -0,0 +1,182 @@
+const addEngineerCard = require("./cards/card-engineer");
+const addInternCard = require("./cards/card-intern");
+const wrapProfileCards = require("./cards/card-wrapper");
+
+//team members array
+const team = [];
+
+const teamNameQuestion = [
+  {
+    name: "teamName",
+    type: "input",
+    message: "What is the name of your team?",
+  },
+];
+
+const managerQuestions = [
+  {
+    type: "input",
+    name: "name",
+    message: "Please enter the Manager's name:",
+  },
+  {
+    type: "input",
+    name: "id",
+    message: "Please enter the Manager's ID number:",
+  },
+  {
+    type: "input",
+    name: "email",
+    message: "Please enter the Manager's Email address:",
+  },
+  {
+    type: "input",
+    name: "officeNumber",
+    message: "Please enter the Manager's Office Number:",
+  },
+  {
+    type: "list",
+    name: "upNext",
+    message: "What would you like to do next?",
+    choices: ["Add Engineer", "Add Intern", "My team is complete!"],
+  },
+];
+
+const engineerQuestions = [
+  {
+    type: "input",
+    name: "name",
+    message: "Please enter Engineer's name:",
+  },
+  {
+    type: "input",
+    name: "id",
+    message: "Please enter Engineer's ID number:",
+  },
+  {
+    type: "input",
+    name: "email",
+    message: "Please enter the Engineer's Email address:",
+  },
+  {
+    type: "input",
+    name: "github",
+    message: "Please enter the Engineer's Github username:",
+  },
+  {
+    type: "list",
+    name: "upNext",
+    message: "What would you like to do next?",
+    choices: ["Add Engineer", "Add Intern", "My team is complete!"],
+  },
+];
+
+const internQuestions = [
+  {
+    type: "input",
+    name: "name",
+    message: "Please enter Intern's name:",
+  },
+  {
+    type: "input",
+    name: "id",
+    message: "Please enter Intern's ID number:",
+  },
+  {
+    type: "input",
+    name: "email",
+    message: "Please enter the Intern's Email address:",
+  },
+  {
+    type: "input",
+    name: "school",
+    message: "Please enter the Intern's school:",
+  },
+  {
+    type: "list",
+    name: "upNext",
+    message: "What would you like to do next?",
+    choices: ["Add Engineer", "Add Intern", "My team is complete!"],
+  },
+];
+
+const askQuestions = async (questionArr) => {
+  inquirer
+    .prompt(questionArr)
+    .then((member) => {
+      team.push(member);
+
+      if (member.upNext === "Add Engineer") {
+        askQuestions(engineerQuestions);
+      } else if (member.upNext === "Add Intern") {
+        askQuestions(internQuestions);
+      } else {
+        createProfiles(team);
+      }
+    })
+    .catch((err) => console.log(err));
+};
+
+const createProfiles = (team) => {
+  const profiles = team.map((member) => {
+    const { name, id, email } = member;
+
+    // If you're adding a manager, ask for office number
+    if (member.hasOwnProperty("officeNumber")) {
+      const { officeNumber } = member;
+      return new Manager(name, id, email, officeNumber);
```

Comment:
When we have a function/class which takes many parameters (generally anything more than 2), it is "safer" to pass these in as an object, rather than separate parameters. For example, here we would do:

`new Manager( { name: name, id: id, email: email, officeNumber: officeNumber })`
or if we implement object shorthand notation:
`new Manager( { name, id, email, officeNumber })`

We would need to change the way we have written the classes to expect an object instead (I will comment this on the class file), but essentially this protects the function/class as it no longer matters in which order we pass the arguments.

The way you currently have it, if I accidentally pass the arguments in the wrong order, it will cause errors, because the class is expecting those parameters separately in that specific order. It assumes whatever is passed in first is the name, even if we accidentally pass the email first.

By passing an object with keys, it is simply looking within that object for a key with a matching value, so we can write the keys in the object in whatever order we like.